src/app/hooks/useWheelZoom.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react';
+
+interface Position {
+  x: number;
+  y: number;
+}
+
+type OnZoom = (delta: number, cursor: Position) => void;
+
+export const useWheelZoom = (element: HTMLElement | null, onZoom: OnZoom) => {
```

Comment:
element should also be of type `() => HTMLElement | null` so we can use it with `useRef` hook.